pandas/tests/extension/test_arrow.py

```diff
@@ -3239,13 +3239,71 @@ def test_arrow_floordiv():
 
 
 def test_arrow_floordiv_large_values():
-    # GH 55561
+    # GH 56645
     a = pd.Series([1425801600000000000], dtype="int64[pyarrow]")
     expected = pd.Series([1425801600000], dtype="int64[pyarrow]")
     result = a // 1_000_000
     tm.assert_series_equal(result, expected)
 
 
+def test_arrow_floordiv_large_integral_result():
+    # GH 56676
+    a = pd.Series([18014398509481983, -9223372036854775808], dtype="int64[pyarrow]")
+    result = a // 1
+    tm.assert_series_equal(result, a)
+
+
+def test_arrow_floordiv_larger_divisor():
```

Comment:
For consistency across all types it would be better if we used the `any_numeric_ea_and_arrow_dtype` fixture in this function (and any function here where the values can be represented across all integral types)

Reply:
any_numeric_ea_and_arrow_dtype includes nullable integer types such as "Int8" which I don't think are affected by this change, should I use tm.ALL_INT_PYARROW_DTYPES_STR_REPR + tm.FLOAT_PYARROW_DTYPES_STR_REPR?